src/test/java/org/gridsuite/study/server/StudyControllerDynamicSecurityAnalysisTest.java

```diff
@@ -0,0 +1,515 @@
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.boot.test.mock.mockito.SpyBean;
+import org.springframework.cloud.stream.binder.test.InputDestination;
+import org.springframework.cloud.stream.binder.test.OutputDestination;
+import org.springframework.http.MediaType;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.support.MessageBuilder;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.BDDMockito.any;
+import static org.mockito.BDDMockito.eq;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+/**
+ * @author Thang PHAM <quyet-thang.pham at rte-france.com>
+ */
+@SpringBootTest
+@AutoConfigureMockMvc
+@DisableElasticsearch
+@ContextConfigurationWithTestChannel
+class StudyControllerDynamicSecurityAnalysisTest {
+    private static final Logger LOGGER = LoggerFactory.getLogger(StudyControllerDynamicSecurityAnalysisTest.class);
+
+    private static final String API_VERSION = StudyApi.API_VERSION;
+    private static final String DELIMITER = "/";
+    private static final String STUDY_END_POINT = "studies";
+
+    private static final String STUDY_BASE_URL = UrlUtil.buildEndPointUrl("", API_VERSION, STUDY_END_POINT);
+    private static final String STUDY_DYNAMIC_SECURITY_ANALYSIS_END_POINT_RUN = "{studyUuid}/root-networks/{rootNetworkUuid}/nodes/{nodeUuid}/dynamic-security-analysis/run";
+    private static final String STUDY_DYNAMIC_SECURITY_ANALYSIS_END_POINT_PARAMETERS = "{studyUuid}/dynamic-security-analysis/parameters";
+    private static final String STUDY_DYNAMIC_SECURITY_ANALYSIS_END_POINT_STATUS = "{studyUuid}/root-networks/{rootNetworkUuid}/nodes/{nodeUuid}/dynamic-security-analysis/status";
+
+    private static final String HEADER_USER_ID_NAME = "userId";
+    private static final String HEADER_USER_ID_VALUE = "userId";
+
+    private static final String VARIANT_ID = "variant_1";
+
+    private static final UUID CASE_UUID = UUID.randomUUID();
```

Comment:
it's better to use fixed UUID

Reply:
Calling randomUUID() to initialize a static value has been used in others tests. It can avoid to write a long uuid string and avoid the conflict with others uuids when using matcher for params while mocking return.